simpleviewinc/keg-hub
# chore(keg-components): Update dependencies

**Ticket**: [ZEN-614](https://jira.simpleviewtools.com/browse/ZEN-614)

## Goal

* Update the dependencies of keg-components
* Fix any issues related to the dependency updates
* Cleanup the build export files
  * Auto create index files and remove the old template files

## Updates
* `repos/keg-components/.storybook/main.js`
  * Commented out `addon-knobs`
    * The package no longer works, and we should update to `addon-controls`
    * The update is outside of this ticket, so I'm jut disabling them for now
* `repos/keg-components/.storybook/preview.js`
  * Updates due to storybook api changes
* `repos/keg-components/configs/aliases.json`
  * Fixed an alias that did not work in some cases
* `repos/keg-components/configs/babel.config.js`
  * Babel config fixes due to dep updates
* `repos/keg-components/configs/eslintrc.config.js`
  * eslint config fixes due to dep updates
* `repos/keg-components/configs/rollup.config.js`
  * Will now update the `react-native` exports to be `react-native-web` when building for web 
  * This will remove the need for the extra alias in `keg-core`, as they will already be correct
  * Moved inputs generation to it's own script
    * This allows the code to be reused when generating the index files
  * Added index generation to work the same as `retheme`
    * No longer need placeholder index files
    * Generates the index files from predefined templates
* `repos/keg-components/configs/webpack.config.js`
  * Minor update due to dep changes
* `repos/keg-components/index.*.js`
  * Removed as they are no longer needed
  * These files now get generated automatically via the `yarn gen:indexes` command
    * This command is also called durring the build process
  * This now works the same as how `retheme` works
* `repos/keg-components/package.json`
  * Updated all the `dependencies` types to the latest versions where needed
  * Updated `peerDependencies` to be ranges
    * This removes a lot of warnings when install `keg-components`, and is also valid
    * Updated which pacakges are optional due to building for web or mobile
  * Add `gen:indexes` command
  * Removed `copy:indexes` command
* `repos/keg-components/src/**/*.stories.mdx`
  * Updated import due to deps updates
* `repos/keg-components/src/components/drawer/drawer.js`
  * Updated to use `forwardRef`
  * Fixed animation issue
* `repos/keg-components/scripts/generateBuildInputs.js`
  * Extracted from `rollup.config.js`
* `repos/keg-components/scripts/generateIndexFiles.js`
  * Copied from `retheme` and modified to work with `keg-component`s
* `repos/keg-components/src/components/modal/modal.stories.mdx`
  * The stories were using the deprecated `knobs` package that was no longer working
  * Had to rewrite the stories with out the `knobs`
  * So I just migrated modal component stories to use `mdx` at the same time
* `repos/keg-components/src/hooks/useFromToAnimation.js`
  * Fixed `useNativeDriver` issue due to `react-native-web` update

## Testing

### Test 1
* Run the command `keg components pack run --package keg-components:zen-614-update-components --env prod`
* This will run the keg-components app, with the keg-components changes of this pr and the retheme changes from this [pr](https://github.com/simpleviewinc/keg-hub/pull/180)
* Navigate to [the keg-components app](http://components-zen-614-update-components.local.keghub.io/) in your browser
* Ensure the the keg-components app works as expected
* Click on all the navigation items
  * Ensure no errors are thrown for every page
  * Ensure the components render as expected

### Test 2
* Run this command `keg dpg run --package keg-core:ZEN-614-update-core --env prod`
  * Then navigate to [the keg-core app](http://core-zen-614-update-core.local.keghub.io/) in your browser
  * This keg-core image is using the changes from this PR 
  * Ensure every thing renders as expected
